device/samples/pnp/pnpTemperatureController.js

```diff
@@ -0,0 +1,256 @@
+  console.log('Sending telemetry message %d from component %s ', index, componentName);
+  
+  const data = JSON.stringify({ temperature: 1 + (Math.random() * 90)});
+  const pnpMsg = new Message(data);
+  if (!!(componentName))
+  {
+    pnpMsg.properties.add(messageSubjectProperty, componentName);
+  }
+  pnpMsg.contentType = 'application/json';
+  pnpMsg.contentEncoding = 'utf-8';
+  await deviceClient.sendEvent(pnpMsg);
+}
+
+async function sendDatasizeTelemetry(deviceClient, componentName, index) {
+  console.log('Sending telemetry message %d from root interface', index);
+  const data = JSON.stringify({ workingset: 1 + (Math.random() * 90)});
+  const pnpMsg = new Message(data);
+  if (!!(componentName))
+  {
+    pnpMsg.properties.add(messageSubjectProperty, componentName);
+  }
+  pnpMsg.contentType = 'application/json';
+  pnpMsg.contentEncoding = 'utf-8';
+  await deviceClient.sendEvent(pnpMsg);
+}
+
+async function main() {
+  // fromConnectionString must specify a transport, coming from any transport package.
+  const client = Client.fromConnectionString(deviceConnectionString, Protocol);
+  console.log('Connecting using connection string ' + deviceConnectionString);
+  let resultTwin;
+
+  try {
+    // Add the modelId here
+    await client.setOptions({ modelId: modelId });
+    await client.open();
+    console.log('Enabling the commands on the client');
+    client.onDeviceMethod(commandNameGetMaxMinReport1, commandHandler);
+    client.onDeviceMethod(commandNameGetMaxMinReport2, commandHandler);
+    client.onDeviceMethod(commandNameReboot, commandHandler);
+
+    // Send Telemetry after some interval
+    let index1 = 0;
+    let index2 = 0;
+    let index3 = 0;
+    intervalToken1 = setInterval(() => {
+      sendTemperatureTelemetry(client, thermostat1ComponentName, index1).catch((err) => console.log('error', err.toString()));
+      index1 += 1;
+    }, 5000);
+
+    intervalToken2 = setInterval(() => {
+      sendTemperatureTelemetry(client, thermostat2ComponentName, index2).catch((err) => console.log('error', err.toString()));
+      index2 += 1;
+    }, 5500);
+
+
+    intervalToken3 = setInterval(() => {
+      sendDatasizeTelemetry(client, null, index3).catch((err) => console.log('error', err.toString()));
+      index3 += 1;
+    }, 6000);
+
+    // attach a standard input exit listener
+    helperAttachExitListener(client);
+
+    try {
+      resultTwin = await client.getTwin();
+      const patchRoot = helperCreateReportedPropertiesPatch({serialNumber:"alohomora"}, null, false);
+      const patchThermostat1Info = helperCreateReportedPropertiesPatch({
+        targetTemperature: {"value": 56.78, "ac": 200, "ad": "wingardium leviosa", "av": 1},
+        maxTempSinceLastReboot: 67.89,
+      }, thermostat1ComponentName);
+
+      const patchThermostat2Info = helperCreateReportedPropertiesPatch({
+        targetTemperature: {"value": 35.67, "ac": 200, "ad": "expecto patronum", "av": 1},
```

Comment:
>expecto patronum" [](start = 62, length = 17)

It's a fun book, but, you MUST change these things.

Reply:
why oh why ? no one cares... the description can be anything right ?